plugins/generator-1.15.2/forge-1.15.2/templates/plant.java.ftl

```diff
@@ -400,6 +400,67 @@ import net.minecraft.block.material.Material;
             </#if>
         </#if>
 
+		<#if (data.canBePlacedOn?size > 0) || hasCondition(data.placingCondition)>
+			<#if data.plantType != "growapable">
+			@Override public boolean isValidGround(BlockState state, IBlockReader worldIn, BlockPos pos) {
+				<#if hasCondition(data.placingCondition)>
+				boolean additionalCondition = true;
+				if (worldIn instanceof IWorldReader) {
+					World world = ((IWorldReader)worldIn).getDimension().getWorld();
+					int x = pos.getX();
+					int y = pos.getY() + 1;
+					int z = pos.getZ();
+					additionalCondition = <@procedureOBJToConditionCode data.placingCondition/>;
+				}
+
+				Block block = state.getBlock();
+
+				</#if>
+				return
+				<#if (data.canBePlacedOn?size > 0)>(
+					<#list data.canBePlacedOn as canBePlacedOn>
+						block == ${mappedBlockToBlockStateCode(canBePlacedOn)}.getBlock()
+						<#if canBePlacedOn?has_next>||</#if>
+					</#list>)
+				</#if>
+				<#if (data.canBePlacedOn?size > 0) && hasCondition(data.placingCondition)> && </#if>
+				<#if hasCondition(data.placingCondition)> additionalCondition </#if>;
+			}
+			</#if>
+
+			@Override public boolean isValidPosition(BlockState state, IWorldReader worldIn, BlockPos pos) {
+				<#if hasCondition(data.placingCondition) && data.plantType == "growapable">
+				World world = worldIn.getDimension().getWorld();
+				int x = pos.getX();
+				int y = pos.getY();
+				int z = pos.getZ();
+				boolean additionalCondition = <@procedureOBJToConditionCode data.placingCondition/>;
+				</#if>
+
+				BlockPos blockpos = pos.down();
+				BlockState blockstate = worldIn.getBlockState(blockpos);
+				Block block = blockstate.getBlock();
+
+				<#if data.plantType = "normal">
+					return this.isValidGround(blockstate, worldIn, blockpos)
+				<#elseif data.plantType == "growapable">
+					return block == this ||

```

Comment:
I see now, there is no isValidGround in case of growapable. Still, additionalCondition is only needed for growapable case, so consider moving that code inside growapable if, as it is not used otherwise. Correct me if I am wrong @AlsoSomeoneElse

Reply:
Indeed there is a check, I missed it. But it is a bit unclean, I will change it a bit, test it, and then wait for another peek at both templates from SomeoneElse before merging ;)